### ISSUE TITLE
"Larger" Font size showing up incorrectly in editor view with Twenty Twenty theme

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#37617]
### Description

The Twenty Twenty theme has opted into the block editor font sizes and styling. Despite this, the "large" font option shows up as bigger than the "larger" font size when selecting between options in the editor. It renders correctly on the front end. Feel free to redirect me if there is a better spot for this :) I tested the same experience with the Twenty Nineteen theme, which has no problem rendering correctly. From what I can gather, the theme is doing everything right to opt into the editor styles and to set the editor font sizes. 

To fix this, I found that if I changed the slug for the "larger" option in twenty twenty to "huge", it worked correctly! This tells me that these font size controls might need to look for more than just "huge" since those options aren't standardized as far as I know. This is just a guess! 

Twenty twenty theme:

```
// Block Editor Font Sizes.
	add_theme_support(
		'editor-font-sizes',
		array(
			array(
				'name'      => _x( 'Small', 'Name of the small font size in the block editor', 'twentytwenty' ),
				'shortName' => _x( 'S', 'Short name of the small font size in the block editor.', 'twentytwenty' ),
				'size'      => 18,
				'slug'      => 'small',
			),
			array(
				'name'      => _x( 'Regular', 'Name of the regular font size in the block editor', 'twentytwenty' ),
				'shortName' => _x( 'M', 'Short name of the regular font size in the block editor.', 'twentytwenty' ),
				'size'      => 21,
				'slug'      => 'normal',
			),
			array(
				'name'      => _x( 'Large', 'Name of the large font size in the block editor', 'twentytwenty' ),
				'shortName' => _x( 'L', 'Short name of the large font size in the block editor.', 'twentytwenty' ),
				'size'      => 26.25,
				'slug'      => 'large',
			),
			array(
				'name'      => _x( 'Larger', 'Name of the larger font size in the block editor', 'twentytwenty' ),
				'shortName' => _x( 'XL', 'Short name of the larger font size in the block editor.', 'twentytwenty' ),
				'size'      => 32,
				'slug'      => 'larger',
			),
		)
	);

	add_theme_support( 'editor-styles' );
```

Twenty Nineteen theme:

```	// Add custom editor font sizes.
		add_theme_support(
			'editor-font-sizes',
			array(
				array(
					'name'      => __( 'Small', 'twentynineteen' ),
					'shortName' => __( 'S', 'twentynineteen' ),
					'size'      => 19.5,
					'slug'      => 'small',
				),
				array(
					'name'      => __( 'Normal', 'twentynineteen' ),
					'shortName' => __( 'M', 'twentynineteen' ),
					'size'      => 22,
					'slug'      => 'normal',
				),
				array(
					'name'      => __( 'Large', 'twentynineteen' ),
					'shortName' => __( 'L', 'twentynineteen' ),
					'size'      => 36.5,
					'slug'      => 'large',
				),
				array(
					'name'      => __( 'Huge', 'twentynineteen' ),
					'shortName' => __( 'XL', 'twentynineteen' ),
					'size'      => 49.5,
					'slug'      => 'huge',
				),
			)
		);
```

### Step-by-step reproduction instructions

1. Activate Twenty Twenty theme
2. Create a new post
3. Try changing around the font sizes based on the provided default options.

### Screenshots, screen recording, code snippet

Video:

https://user-images.githubusercontent.com/26996883/147261978-e09913a4-d8ca-4ddf-a53d-881fb6b64e0b.mov

Twenty Twenty Editor view:

<img width="682" alt="Screen Shot 2021-12-23 at 10 33 07 AM" src="https://user-images.githubusercontent.com/26996883/147261565-56731700-969a-44bf-9e3d-74d50e3ad682.png">

Twenty TwentyFront end view: 
<img width="688" alt="Screen Shot 2021-12-23 at 10 33 01 AM" src="https://user-images.githubusercontent.com/26996883/147261569-21b14b29-5547-4e50-9d80-1f0a2a1e0054.png">

For fun, I tried changing around the front size to see if something was acting up there and noticed if I changed the font size to just 26, it showed up differently in the sidebar, listing out each number:

<img width="1199" alt="Screen Shot 2021-12-23 at 10 28 20 AM" src="https://user-images.githubusercontent.com/26996883/147261621-43b98022-cc03-4f60-9bb5-004304f66e76.png">

I noticed that rather than showing the actual numbers too, the options show up as "1, 2, 3, 4" - wasn't sure if that was intentional and wanted to note as a result:

<img width="1218" alt="Screen Shot 2021-12-23 at 10 28 01 AM" src="https://user-images.githubusercontent.com/26996883/147261686-179dd75b-037a-4158-b944-8c6e19595554.png">

Here's Twenty Twenty working correctly after setting the "larger" font size slug to "huge":

<img width="1169" alt="Screen Shot 2021-12-23 at 10 47 43 AM" src="https://user-images.githubusercontent.com/26996883/147263256-fbccb959-366e-43a3-a94e-9ee965f0d982.png">

### Environment info

- WordPress 5.9-beta4-52409 using the beta tester plugin with a nightly build
- Gutenberg not active
- Twenty Twenty theme

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by oandregal** [comment]
I gave this a quick test and it seems the CSS Custom Properties for the font sizes provided by the theme are not being generated (they should), hence, the values fall back to the ones provided by core.

I've also noticed that the `small` font size had `13px` (default) and not `18px` as set by the theme.

**Comment by neilorangepeel** [comment]
I'm working on a custom theme and the 'normal' and 'large' CSS Custom Properties are being generated but are being overwritten by the [editor.scss](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/editor.scss) as it has a higher specificity. `:root .editor-styles-wrapper ` vs `.editor-styles-wrapper `.

<img width="357" alt="Screenshot 2021-12-28 at 17 47 15" src="https://user-images.githubusercontent.com/3746751/147617369-e4cabf15-28ca-4b0d-928f-49df60426ecb.png">

**Comment by chthonic-ds** [comment]
@neilorangepeel https://github.com/WordPress/gutenberg/pull/37526 appears to lower the specificity and correct the issue you're seeing.

**Comment by Mamaduka** [comment]
Hi, @oandregal 

Should we fix this for 5.9, or can it be punted to a minor release?

**Comment by oandregal** [comment]
Related to this issue, I've found this other one and prepared a fix https://github.com/WordPress/gutenberg/pull/37819

**Comment by oandregal** [comment]
Also related to this issue: https://github.com/WordPress/wordpress-develop/pull/2129 Apparently, this piece of logic wasn't backported.

**Comment by oandregal** [comment]
The final PR I've prepared to fix this issue is this https://github.com/WordPress/wordpress-develop/pull/2130

**Comment by oandregal** [comment]
With the 3 PRs above this would be fixed. Let me unwrap what happened. The issue here is that TwentyTwentyTwenty hasn't been updated to set the default presets via CSS Custom Properties (in lieu of having a `theme.json`), see the "Changes to the global stylesheet" in [this devnote](https://make.wordpress.org/core/2022/01/08/updates-for-settings-styles-and-theme-json/).

So:

- https://github.com/WordPress/gutenberg/pull/37819 lowers the specificity for legacy fonts, those that are no longer WordPress defaults, so we're a bit more friendly to themes.
- https://github.com/WordPress/wordpress-develop/pull/2129 fixes WordPress 5.9 so the global stylesheet is loaded for all themes (something happened during the backports that we missed this).
- https://github.com/WordPress/wordpress-develop/pull/2130 updates TwentyTwenty to set the default presets via CSS Custom Properties.

**Comment by noisysocks** [comment]
It looks like we can close this as all three of those PRs have been merged.
